src/logcollector/read_win_el.c

```diff
@@ -509,6 +509,20 @@ void readel(os_el *el, int printit)
         }
     }
 
+
+    /* Event log was closed and re-opened */
+    else if (id == ERROR_INVALID_HANDLE) {
+        mdebug1("EventLog was closed and re-opened");
+
+        CloseEventLog(el->h);
+        el->h = NULL;
+
+        /* Reopen */
+        if (startEL(el->name, el) < 0) {
+            merror("Unable to reopen event log '%s'", el->name);
+        }
+    }
+
     else {
         mdebug1("Error reading event log: %d", id);
```

Comment:
In my environment (Windows Server 2016) the code flow is always coming in here giving error 183. Despite this, Eventlog events are captured and sent to the manager as expected.

However, if I restart the service, it never reconnected. We must investigate this use case.

Reply:
I've been trying to reproduce this (Windows Server 2016) and I haven't been able to.
I stopped the EventLog Service by executing this commands:

`Stop-Service EventLog -Force`
`taskkill /pid PID /f` (Start EventLog manually after that)

On both cases, Wazuh reconnects with EventLog successfully, on the first case the error 1717 is prompted until the service is reconnected. In the second case is the error 1722 which is prompted.

Reply:
It's not a bug, it's a false positive produced by an error when debugging the functionality, which was changing the error code returned by `ReadEventLog` function. The solution is working correctly for Eventlog.